BrainPortal/app/models/message.rb

```diff
@@ -329,4 +354,24 @@ def self.parse_markup(string) #:nodoc:
     arr.join.html_safe
   end
 
+  # before_destroy callback
+  def track_resource_usage_create
+    return true if self.message_type != "communication"  # only user to user communication is counted so far
+    CountResourceUsageForUserMessage.create(
+      :value              => 1,
+      :user_id            => self.sender_id
+    )
+    true
+  end
+
+  # before_destroy callback
```

Comment:
We don't need to track destroy events. please remove this callback

Reply:
Why not? Do we have a way to see how Delete button being used.